### PR TITLE
[WFLY-14307] Warning if wrong annotation is added to a Servlet

### PR DESCRIPTION
**Pull request by gaol**
Thanks for submitting your Pull Request!

Please delete this text, and add a link to the Jira issue solved by this PR.

If this PR is not for the 'main' branch you must add a link to the equivalent change in 'main'.

Remember to use the Jira issue ID in the PR title and any commits.

Issue: https://issues.redhat.com/browse/WFLY-14307

### CODE REVIEW

**Comment by gaol** [comment]
@fl4via would you please review? thanks

File: undertow/src/main/java/org/wildfly/extension/undertow/logging/UndertowLogger.java [hunk 1]
```diff
@@ -442,4 +442,8 @@ public interface UndertowLogger extends BasicLogger {
     @Message(id = 110, value = "The use of security realms at runtime is unsupported.")
     OperationFailedException runtimeSecurityRealmUnsupported();
 
+    @LogMessage(level = WARN)
+    @Message(id = 111, value = "Annotation: '%s' cannot be used on Servlet: '%s'")
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/logging/UndertowLogger.java:446]: This should say something about what will happen, e.g. the annotation will have no effect (if that's true.)

**Comment by gaol** [comment]
@bstansberry Updated with the suggestion :)

File: undertow/src/main/java/org/wildfly/extension/undertow/logging/UndertowLogger.java (hunk 1 above)
Comment (bstansberry) [reply undertow/src/main/java/org/wildfly/extension/undertow/logging/UndertowLogger.java]: Thanks.

**Comment by soul2zimate** [comment]
@bstansberry  @fl4via Is this good to merge ?  I'd like to have an approval so I can merge the downstream one.

**Review by fl4via — APPROVED** [review]

**Comment by fl4via** [comment]
> @bstansberry @fl4via Is this good to merge ? I'd like to have an approval so I can merge the downstream one.

@soul2zimate yes, it is good to go! We probably need to retrigger the cancelled CI though to make sure it passes.

**Comment by soul2zimate** [comment]
retest this please

**Comment by soul2zimate** [comment]
It seems I only triggered jobs configured in TeamCity, but I am not able to rerun that Github action`WildFly Build / WildFly Build (windows-latest) (pull_request)`

**Comment by bstansberry** [comment]
Thanks @gaol, @fl4via and @soul2zimate

**Pull request merged**